crates/ironrdp-web/src/session.rs

```diff
@@ -492,18 +477,122 @@ impl Session {
+                            let scaled_height = cast_f64_to_u16_lossy(f64::from(pointer.height) * scale);
+                            let hotspot_x = cast_f64_to_u16_lossy(f64::from(pointer.hotspot_x) * scale);
+                            let hotspot_y = cast_f64_to_u16_lossy(f64::from(pointer.hotspot_y) * scale);
+
+                            // Per invariants: scaled_width * scaled_height * 4 <= 32 * 32 * 4 < usize::MAX
+                            #[allow(clippy::arithmetic_side_effects)]
+                            let resized_rgba_buffer_size = usize::from(scaled_width * scaled_height * 4);
+
+                            let mut rgba_resized = vec![0u8; resized_rgba_buffer_size];
+                            let mut resizer = resize::new(
+                                usize::from(pointer.width),
+                                usize::from(pointer.height),
+                                usize::from(scaled_width),
+                                usize::from(scaled_height),
+                                resize::Pixel::RGBA8P,
+                                resize::Type::Lanczos3,
+                            )
+                            .context("failed to initialize cursor resizer")?;
+
+                            resizer
+                                .resize(pointer.bitmap_data.as_pixels(), rgba_resized.as_pixels_mut())
+                                .context("failed to resize cursor")?;
+
+                            (
+                                scaled_width,
+                                scaled_height,
+                                hotspot_x,
+                                hotspot_y,
+                                Cow::Owned(rgba_resized),
+                            )
+                        } else {
+                            (
+                                pointer.width,
+                                pointer.height,
+                                pointer.hotspot_x,
+                                pointer.hotspot_y,
+                                Cow::Borrowed(pointer.bitmap_data.as_slice()),
+                            )
+                        };
+
+                        // Encode PNG.
+                        let mut png_buffer = Vec::new();
+                        {
+                            let mut encoder =
+                                png::Encoder::new(&mut png_buffer, u32::from(png_width), u32::from(png_height));
+
+                            encoder.set_color(png::ColorType::Rgba);
+                            encoder.set_depth(png::BitDepth::Eight);
+                            let mut writer = encoder.write_header().context("PNG encoder header write failed")?;
+                            writer
+                                .write_image_data(rgba_buffer.as_ref())
+                                .context("failed to encode pointer PNG")?;
+                        }
```

Comment:
Question: could we use `image/bmp` instead, and avoid the cost of converting to a PNG?

Reply:
Yeah, we could use it, but:
- `bmp` crate API very sketchy, it preallocates the filled buffer first and then requires a call to set each pixel on specific coordinates separately. 
- Base64-encoded representation will be bigger in size

I would rather stick with PNG, I don't think we will win a lot of performance by switching to BMP. The only thing which I really agree should be added is to set PNG compression type explicitly to `png::Compression::Fast`

Reply:
Ah, you mean for the header? That’s fair. You could probably just use the definition we have in `ironrdp-cliprdr-format` crate, but  I understand your point too. Sounds good to me.